analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/LoopDetector.cs

```diff
@@ -83,13 +83,22 @@ void MergeWithIntersectingLoops(int[] path, int last)
         }
     }
 
-    private static IEnumerable<int> Successors(ControlFlowGraph cfg, int block)
+    private static IEnumerable<int> Successors(ControlFlowGraph cfg, int index)
     {
-        var successors = cfg.Blocks[block].SuccessorBlocks.Select(x => x.Ordinal);
-        if (cfg.Blocks[block].EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
+        var block = cfg.Blocks[index];
+        var successors = block.SuccessorBlocks.Select(x => x.Ordinal);
+        if (block.EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
         {
             successors = successors.Concat(tryRegion.ReachableHandlers().Select(x => x.FirstBlockOrdinal));
         }
+        if (block.EnclosingRegion(ControlFlowRegionKind.Finally) is { } finallyRegion)
+        {
+            var tryFinally = finallyRegion.EnclosingRegion;
+            successors = successors.Concat(
```

Comment:
This needs to work with `tryRegion` only, not `tryFinally`, because that one adds also all blocks from the `finally`
```
var tryRegion = tryFinally.NestedRegion(ControlFlowRegionKind.Try);
```

Reply:
The change you propose makes sense. Just for educational reasons: Is there a scenario where this could potentially lead to a problem?

Reply:
No, the filtering condition that followed would kill all the useless candidates.
The problem here is just the confusion and performance.